File: Desktop/EX4/src/client/src/Checker.cpp
#include "../include/Checker.h"

using namespace std;

Checker::Checker() {

    //Initialize the checker:
    this->setEmpty();
}

char Checker::getVal() const {
    return val;
}

void Checker::setX() {
    val = 'X';
}

void Checker::setO() {
    val = 'O';
}

void Checker::setEmpty() {
    val = ' ';
}

bool Checker::isEmpty() const {
    return (val == ' ');
}

void Checker::setPosition(int row, int col) {
    pos.row = row;
    pos.col = col;
}

struct Checker::position Checker::getPosition() const {
    return pos;
}

File: src/logic/Checker.java
package logic;

public class Checker {
    private Position pos;
    private char val;

    public Checker() {
        this.pos = new Position(0, 0);
        this.setEmpty();
    }

    // Methods

    public char getVal() {
        return this.val;
    }

    public void setX() {
        this.val = 'X';
    }

    public void setO() {
        this.val = 'O';
    }

    private void setEmpty() {
        val = ' ';
    }

    public boolean isEmpty() {
        return (this.val == ' ');
    }

    public void setPosition(int row, int col) {
        this.pos.row = row;
        this.pos.col = col;
    }

    public Position getPosition() {
        return this.pos;
    }
}
